### PR TITLE
WIP: Feature/v3 configuration

### PR DESCRIPTION
**Pull request by jaapio**
This pr is a follow up on pr #1940. 
It replaces the configuration of phpdocumentor with the new v3 config setup. A lot of refactoring was needed to accomplish this. I moved the commands to stages of a pipeline, the commands are refactored to pipeline only.

### CODE REVIEW

**Comment by jaapio** [comment]
This pr is blocked by some internal discussions and tests of a new plugin structure.

**Comment by ashnazg** [comment]
If there's a `.env.dist`, does that imply that the `.env` file should not be committed in?

File: src/phpDocumentor/Parser/Middleware/EmittingMiddleware.php [hunk 1]
```diff
@@ -26,6 +26,7 @@
      */
     public function execute($command, callable $next)
     {
+        var_dump($command->getFile()->path());
```
Comment (ashnazg) [inline src/phpDocumentor/Parser/Middleware/EmittingMiddleware.php]: `var_dump()` intentional?

Comment (jaapio) [reply src/phpDocumentor/Parser/Middleware/EmittingMiddleware.php]: Saw that one to. Fixing it currently :smiley:

**Pull request merged**